browser/js/content.js

```diff
@@ -302,28 +299,54 @@ function submitChat(successCallback, e) {
       return;
     }
 
-    removeChat(f, true /* skipCheck */);
+    // Success!
     nudgeCallback();
     successCallback !== undefined && successCallback();
+    installCopypasta(f.getElementsByClassName("keybase-copy"));
   });
 }
 
+// Render a success screen which replaces the body of the widget.
+function renderSuccess(el, closeCallback, extraHTML) {
+  el.innerHTML = `
+    <h3>
+      <img src="${asset("images/icon-keybase-logo-16.png")}"
+           srcset="${asset("images/icon-keybase-logo-16@2x.png")} 2x, ${asset("images/icon-keybase-logo-16@3x.png")} 3x"
+           />
+      Keybase chat <span class="keybase-close"> </span>
+    </h3>
+    <div class="keybase-body">
+      <p>
+        <img src="${asset("images/icon-fancy-chat-72-x-52.png")}" style="width: 72px; height: 52px;"
+             srcset="${asset("images/icon-fancy-chat-72-x-52@2x.png")} 2x, ${asset("images/icon-fancy-chat-72-x-52@3x.png")} 3x"
+             />
+      </p>
+      <p>
+        Chat sent! You can continue the coversation in your Keybase app.
+      </p>
+      ${extraHTML || ""}
+      <p>
+        <input type="button" class="keybase-close" value="Close" />
+      </p>
+    </div>
+  `;
+  el.className = "keybase-success";
+
+  installCloser(el.getElementsByClassName("keybase-close"), el, true /* skipCheck */, closeCallback);
+}
+
 // Render an error that replaces the body of the widget.
 function renderErrorFull(el, bodyHTML) {
   el.innerHTML = `
     <h3><span class="keybase-close"> </span></h3>
     <p>
-      <img src="${chrome.runtime.getURL("images/icon-keybase-logo-128.png")}" style="height: 64px; width: 64px;" />
+      <img src="${asset("images/icon-keybase-logo-128.png")}" style="height: 64px; width: 64px;" />
     </p>
     ${bodyHTML}
   `;
   el.className = "keybase-error";
-  //
-  // Install closing button (the "x" in the corner)
-  const closer = el.getElementsByClassName("keybase-close")[0];
-  closer.addEventListener("click", function(e) {
-    removeChat(el, true /* skipCheck */);
-  });
+
+  installCloser(el.getElementsByClassName("keybase-close"), el, true /* skipCheck */);
```

Comment:
I generally always leave a comment for naked boolean values, regardless of how many params there are. It's hard to infer what they mean otherwise.

Otherwise, I feel like it's not quite there in terms of arg complexity. I did end up pulling out the User stuff into its own object type because it was getting too complicated.